src/Autofac/RegistrationExtensions.cs

```diff
@@ -79,21 +84,6 @@ public static IRegistrationBuilder<T, SimpleActivatorData, SingleRegistrationSty
 
             activator.DisposeInstance = rb.RegistrationData.Ownership == InstanceOwnership.OwnedByLifetimeScope;
 
-            // https://github.com/autofac/Autofac/issues/1102
-            // Single instance registrations with any custom activation phases (i.e. activation handlers) need to be auto-activated,
-            // so that other behavior (such as OnRelease) that expects 'normal' object lifetime behavior works as expected.
-            if (rb.ResolvePipeline.Middleware.Any(s => s.Phase == PipelinePhase.Activation))
-            {
-                var autoStartService = rb.RegistrationData.Services.First();
```

Comment:
The bug here is due to the possibility that there are multiple registrations for this service; resolving it won't guarantee that we end up with the same component (and in fact, some unrelated random thing could end up unexpectedly activated :-))